### PR TITLE
Use player response for parsing

### LINKED ISSUE #177: Rework videoinfo parser to use player_response
YouTube seems to be slowly removing fields from videoinfo but expanding player_response JSON data. It already contains everything that is (or was) available otherwise. Need to rewrite the parser to make use of that.

### PR DESCRIPTION
**Pull request by Tyrrrz**
Closes #177

### CODE REVIEW

**Comment by Tyrrrz** [comment]
@SlowLogicBoy check this out, signature deciphering is no longer needed with this

File: YoutubeExplode.Tests/YoutubeClientIntegrationTests.cs [hunk 1]
```diff
@@ -87,15 +87,6 @@ public void YoutubeClient_GetVideoMediaStreamInfosAsync_Unavailable_Test(string
             Assert.ThrowsAsync<VideoUnavailableException>(() => client.GetVideoMediaStreamInfosAsync(videoId));
         }
 
-        [Test]
-        [TestCaseSource(typeof(Data), nameof(Data.GetVideoIds_RequiresPurchase))]
-        public void YoutubeClient_GetVideoMediaStreamInfosAsync_RequiresPurchase_Test(string videoId)
```
Comment (SlowLogicBoy) [inline YoutubeExplode.Tests/YoutubeClientIntegrationTests.cs]: What is happening with these now?

**Comment by SlowLogicBoy** [comment]
Looks like they are getting rid of their legacy code. and obsoleting old stuff.
Wow there are a lot of changes, looks scary, but code base is so much smaller 😆 performance should increase quite a bit.

File: YoutubeExplode/YoutubeClient.Video.cs [hunk 2]
```diff
@@ -185,180 +136,138 @@ public async Task<MediaStreamInfoSet> GetVideoMediaStreamInfosAsync(string video
...
+                var bitrate = streamInfoParser.ParseBitrate();
+                var format = streamInfoParser.ParseFormat();
+                var audioEncoding = streamInfoParser.ParseAudioEncoding();
+                var videoEncoding = streamInfoParser.ParseVideoEncoding();
+                var videoQualityLabel = streamInfoParser.ParseQualityLabel();
+                var videoQuality = VideoQualityConverter.FromVideoQualityLabel(videoQualityLabel);
+                var width = streamInfoParser.ParseWidth();
+                var height = streamInfoParser.ParseHeight();
+                var resolution = new VideoResolution(width, height);
+                var framerate = 25; // there is no info about framerate for muxed streams
```
[the remark this answers is not in the document]
Comment (Tyrrrz) [reply YoutubeExplode/YoutubeClient.Video.cs]: I'll take a look into it, this is temporary

File: YoutubeExplode.Tests/YoutubeClientIntegrationTests.cs (hunk 1 above)
Comment (Tyrrrz) [reply YoutubeExplode.Tests/YoutubeClientIntegrationTests.cs]: They are still unplayable and detectable, but there's no preview video ID anymore. I will look into reworking exceptions into Unavailable and Unplayable. Unavailable videos are those that don't exist, are deleted or private (maybe some other cases), unplayable videos are those that need purchase, those that are controversial or banned in your country (also maybe some other cases). Only for video streams it matters that the video should be playable.

**Comment by Tyrrrz** [comment]
The API is a lot smoother now. They've added tons of info for Muxed streams though, but it seems that it still is inconsistent - sometimes it's not there. I will wait with putting extra info into models until whatever they're doing becomes more stable.

**Pull request merged**